libs/@local/hash-graph-types/rust/src/knowledge/entity/provenance.rs

```diff
@@ -7,9 +7,135 @@ use bytes::BytesMut;
 use postgres_types::{FromSql, IsNull, Json, ToSql, Type};
 use serde::{Deserialize, Serialize};
 use temporal_versioning::{DecisionTime, Timestamp, TransactionTime};
+use time::OffsetDateTime;
+use url::Url;
 
 use crate::account::{CreatedById, EditionArchivedById, EditionCreatedById};
 
+#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
+#[cfg_attr(feature = "utoipa", derive(utoipa::ToSchema))]
+#[serde(deny_unknown_fields, rename_all = "camelCase")]
+pub struct Tool {
+    /// The name of the tool.
+    pub name: String,
+
+    /// The tool version, in whatever format the tool natively provides.
+    #[cfg_attr(feature = "utoipa", schema(nullable = false))]
+    #[serde(default, skip_serializing_if = "Option::is_none")]
+    pub version: Option<String>,
+
+    /// The tool version in the format specified by Semantic Versioning 2.0.
+    #[cfg_attr(feature = "utoipa", schema(nullable = false, value_type = String))]
+    #[serde(default, skip_serializing_if = "Option::is_none")]
+    pub semantic_version: Option<semver::Version>,
+
+    /// The organization or company that produced the tool.
+    #[cfg_attr(feature = "utoipa", schema(nullable = false))]
+    #[serde(default, skip_serializing_if = "Option::is_none")]
+    pub organization: Option<String>,
+}
+
+#[derive(Debug, Copy, Clone, PartialEq, Eq, Serialize, Deserialize)]
+#[cfg_attr(feature = "utoipa", derive(utoipa::ToSchema))]
+#[serde(deny_unknown_fields, rename_all = "kebab-case")]
+pub enum SourceType {
+    Webpage,
+    Document,
+}
```

Comment:
Discussed offline that we're keeping this as an enum and add fields as we need them. TBD: Add docs how to generate the OpenAPI specs for it.